core/panic-handler/src/lib.rs

```diff
@@ -19,20 +19,51 @@
 use backtrace::Backtrace;
 use std::io::{self, Write};
 use std::panic::{self, PanicInfo};
+use std::cell::Cell;
 use std::thread;
 
+thread_local! {
+	pub static ABORT: Cell<bool> = Cell::new(true);
+}
+
 /// Set the panic hook
-pub fn set() {
-	panic::set_hook(Box::new(panic_hook));
+pub fn set(bug_url: &'static str) {
+	panic::set_hook(Box::new(move |c| panic_hook(c, bug_url)));
 }
 
-static ABOUT_PANIC: &str = "
+macro_rules! ABOUT_PANIC {
+	() => ("
 This is a bug. Please report it at:
 
-    https://github.com/paritytech/polkadot/issues/new
-";
+	{}
+")}
+
+/// Set aborting flag. Returns previous value of the flag.
+pub fn set_abort(enabled: bool) -> bool {
```

Comment:
I think this function should not be public?

Reply:
panics in the runtime should not abort, but panics in client code, including exernalities, should.

Reply:
@bkchr I'd leave it public in case someone needs to set and forget. The crate API aims to be universal.